hdf-eos5: Fix issue when linking against hdf5+szip (#23411)

payerle opened the pull request:
Should fix issue #23411 when linking against hdf5+szip

Also fix bug if hdf5 does not depend on zlib

Reluctantly added payerle as a maintainer

payerle committed:
hdf-eos5: Fix issue when linking against hdf5+szip (#23411)

Should fix issue #23411 when linking against hdf5+szip

Also fix bug if hdf5 does not depend on zlib

Reluctantly added payerle as a maintainer

alalazo commented on var/spack/repos/builtin/packages/hdf-eos5/package.py:
```diff
@@ -46,6 +46,11 @@ class HdfEos5(AutotoolsPackage):
     conflicts('~static', when='~shared',
               msg='At least one of +static or +shared must be set')
 
+    # payerle: I am adding myself as a maintainer because I initially wrote
+    # the spack recipe, but I am not really an user of the code, so please
+    # consider becoming a maintainer if you use this code.
```
I think this is unnecessary as a comment in the package. Maybe when somebody modifies the package you can comment in this direction in the PR?

alalazo reviewed (commented):
Left a minor comment, otherwise LGTM.

sethrj commented on var/spack/repos/builtin/packages/hdf-eos5/package.py:
```diff
@@ -92,7 +97,10 @@ def configure_args(self):
         # Provide config args for dependencies
         extra_args.append('--with-hdf5={0}'.format(self.spec['hdf5'].prefix))
-        if self.spec['zlib']:
+        if 'libszip' in self.spec:
```
I've recently learned that this (and below) should probably be the virtual  package `szip`, of which `libszip` is the default implementation but `libaec` is an alternative. See the discussion in https://github.com/spack/spack/pull/18937 .

payerle committed:
[hdf-eos5]: cleanup as suggested by @sethrj

Removed extraneous comments
Changed check for libszip dependency to check for szip dependency
